napari/plugins/_hook_callers.py

```diff
@@ -177,51 +158,240 @@ def _multicall(
+            ``self.get_hookimpls()``, but those that are not included
+            will be left at the end of the call order.
+
+        Raises
+        ------
+        TypeError
+            If any item in ``new_order`` is neither a string (plugin_name) or a
+            ``HookImpl`` instance.
+        ValueError
+            If any item in ``new_order`` is neither the name of a plugin or a
+            ``HookImpl`` instance that is present in self._nonwrappers.
+        ValueError
+            If ``new_order`` argument has multiple entries for the same
+            implementation.
+
+        Examples
+        --------
+        Imagine you had a hook specification named ``print_plugin_name``, that
+        expected plugins to simply print their own name. An implementation
+        might look like:
+
+        >>> # hook implementation for ``plugin_1``
+        >>> @hook_implementation
+        ... def print_plugin_name():
+        ...     print("plugin_1")
+
+        If three different plugins provided hook implementations. An example
+        call for that hook might look like:
+
+        >>> plugin_manager.hook.print_plugin_name()
+        plugin_1
+        plugin_2
+        plugin_3
+
+        If you wanted to rearrange their call order, you could do this:
+
+        >>> new_order = ["plugin_2", "plugin_3", "plugin_1"]
+        >>> plugin_manager.hook.print_plugin_name.bring_to_front(new_order)
+        >>> plugin_manager.hook.print_plugin_name()
+        plugin_2
+        plugin_3
+        plugin_1
+
+        You can also just specify one or more item to move them to the front
+        of the call order:
+        >>> plugin_manager.hook.print_plugin_name.bring_to_front(["plugin_3"])
+        >>> plugin_manager.hook.print_plugin_name()
+        plugin_3
+        plugin_2
+        plugin_1
+        """
+        # make sure items in order are unique
+        if len(new_order) != len(set(new_order)):
+            raise ValueError("repeated item in order")
+
+        # make new lists for the rearranged _nonwrappers
+        # for details on the difference between wrappers and nonwrappers, see:
+        # https://pluggy.readthedocs.io/en/latest/#wrappers
+        _old_nonwrappers = self._nonwrappers.copy()
+        _new_nonwrappers: List[HookImpl] = []
+        indices = [self.index(elem) for elem in new_order]
+        for i in indices:
+            _new_nonwrappers.insert(0, _old_nonwrappers[i])
+
+        # remove items that have been pulled, leaving only items that
+        # were not specified in ``new_order`` argument
+        # do this rather than using .pop() above to avoid changing indices
+        for i in sorted(indices, reverse=True):
+            del _old_nonwrappers[i]
+
+        # if there are any hook_implementations left over, add them to the
+        # beginning of their respective lists
+        if _old_nonwrappers:
+            _new_nonwrappers = [x for x in _old_nonwrappers] + _new_nonwrappers
```

Comment:
because [they](https://github.com/napari/napari/pull/937/files/a7c7da78fa1f68bc6de9fe0e669d17fbb5762d11#diff-a034bf860a5ae6372e719ef0f973224f) [get called](https://github.com/napari/napari/pull/1023#discussion_r392614235) in [reverse order](https://github.com/napari/napari/blob/1ae61bbb428ddf4eeef57aebc89bc6474bcb7751/napari/_qt/qt_plugin_sorter.py#L152-L154) :)

will add a comment.